hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -288,12 +288,9 @@ public static <R> HoodieData<HoodieRecord<R>> mergeForPartitionUpdatesIfNeeded(
     // the tagging partitions and locations
     // NOTE: The incoming records may only differ in record position, however, for the purpose of
     //       merging in case of partition updates, it is safe to ignore the record positions.
-    HoodieData<HoodieRecordGlobalLocation> globalLocations = incomingRecordsAndLocations
+    HoodieData<Pair<String, String>> globalLocations = incomingRecordsAndLocations
         .filter(p -> p.getRight().isPresent())
-        .map(p -> new HoodieRecordGlobalLocation(
-            p.getRight().get().getPartitionPath(),
-            p.getRight().get().getInstantTime(),
-            p.getRight().get().getFileId()))
+        .map(p -> Pair.of(p.getRight().get().getPartitionPath(), p.getRight().get().getFileId()))
```

Comment:
Wondering if we can simply remove `instantTime` from the `equals` and `hashcode` of `HoodieRecordGlobalLocation`. Why would we need `instantTime` for global location? Clustering will anyway map to a new fileId. Compaction will create a base file with same file id but different instant time. However, for the location all we care about is partition path and filegroup id right. Instant time is only used for filesystem view.

Reply:
I guess this is a legacy design, to give a quick access of the instant time from a given Hoodie record, but I agree with you, the location should not include the instant time. The `HoodieRecordLocation` should also be fixed.

Reply:
instant time is used in other places as well (and not just global indexing). I do not want to include that in this patch. For now, lets proceed w/ 0.15.0 with cur fix and work towards properly fixing the RecordLocation.

Reply:
yeah let's track separately.

Reply:
https://issues.apache.org/jira/browse/HUDI-7800

Reply:
https://issues.apache.org/jira/browse/HUDI-7800